core/src/main/java/com/linecorp/armeria/common/encoding/ZlibStreamDecoder.java

```diff
@@ -17,14 +17,29 @@
 package com.linecorp.armeria.common.encoding;
 
 import io.netty.buffer.ByteBufAllocator;
-import io.netty.handler.codec.compression.ZlibCodecFactory;
+import io.netty.handler.codec.compression.JZlibDecoder;
+import io.netty.handler.codec.compression.JdkZlibDecoder;
+import io.netty.handler.codec.compression.ZlibDecoder;
 import io.netty.handler.codec.compression.ZlibWrapper;
+import io.netty.util.internal.SystemPropertyUtil;
 
 /**
  * A {@link StreamDecoder} that uses zlib ('gzip' or 'deflate').
  */
 final class ZlibStreamDecoder extends AbstractStreamDecoder {
-    ZlibStreamDecoder(ZlibWrapper zlibWrapper, ByteBufAllocator alloc) {
-        super(ZlibCodecFactory.newZlibDecoder(zlibWrapper), alloc);
+
+    private static final boolean noJdkZlibDecoder =
+            SystemPropertyUtil.getBoolean("io.netty.noJdkZlibDecoder", false);
+
+    ZlibStreamDecoder(ZlibWrapper zlibWrapper, ByteBufAllocator alloc, int maxLength) {
+        super(newZlibDecoder(zlibWrapper, maxLength), alloc, maxLength, false);
+    }
+
+    private static ZlibDecoder newZlibDecoder(ZlibWrapper wrapper, int maxLength) {
+        if (noJdkZlibDecoder) {
+            return new JZlibDecoder(wrapper, maxLength);
+        } else {
+            return new JdkZlibDecoder(wrapper, true, maxLength);
```

Comment:
What's the `true` mean? It seems like the default value is `false` in `JdkZkipDecoder`.

Reply:
The original API we used `ZlibCodecFactory.newZlibDecoder()` internally set `decompressConcatenated` to true.
https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/compression/ZlibCodecFactory.java#L124

Reply:
https://github.com/netty/netty/commit/6ff48dcbe311d2736249b02b31a8745e719d1304
It seems like the Gzip stream could be concatenation of multiple streams.